apps/judicial-system/backend/src/app/modules/file/file.service.ts

```diff
@@ -233,4 +235,33 @@ export class FileService {
 
     return { success }
   }
+
+  async updateCaseFile(
+    caseId: string,
+    fileId: string,
+    update: { [key: string]: string | null },
+    transaction?: Transaction,
+  ): Promise<void> {
+    const promisedUpdate = transaction
+      ? this.fileModel.update(update, {
+          where: { id: fileId, caseId },
+          transaction,
+        })
+      : this.fileModel.update(update, {
+          where: { id: fileId, caseId },
+        })
+
+    const [numberOfAffectedRows] = await promisedUpdate
+
+    if (numberOfAffectedRows > 1) {
+      // Tolerate failure, but log error
+      this.logger.error(
+        `Unexpected number of rows (${numberOfAffectedRows}) affected when updating file ${fileId} of case ${caseId}`,
+      )
+    } else if (numberOfAffectedRows < 1) {
+      throw new InternalServerErrorException(
+        `Could not update file ${fileId} of case ${caseId}`,
+      )
+    }
```

Comment:
Do we want to return `{ success }` here like the other methods in the service?

Reply:
Thank you for pointing out the discrepancy. Actually, there are other methods that do not return `{ success }`, but all the other methods return something. The reason I did not do that here is that this method's return value will not be passed to the client and is not used by the caller (in this case the `archive` method in the `CaseService`. However, to stick to the patterned used elsewhere on the server, this method should return the updated case file. I have made that change.